.buildkite/prepare.sh

```diff
@@ -6,20 +6,20 @@ set -x
 
 "$SOURCE_DIR/.buildkite/replace-vespa-version-in-poms.sh" "$VESPA_VERSION" "$SOURCE_DIR"
 
+cd $SOURCE_DIR
+find . -name "*.java" |
+    awk -F/ '{print $2}' | sort -u > /tmp/list.java-modules.txt
+find . -name "package-info.java" -print0 | xargs -0 grep -HnE "@(com.yahoo.api.annotations.)?PublicApi.*"  |
+    awk -F/ '{print $2}' | sort -u > /tmp/list.public-api-modules.txt
+
 # We disable javadoc for all modules not marked as public API
-for MODULE in $(comm -2 -3 \
-                <(find . -name "*.java" | awk -F/ '{print $2}' | sort -u)
-                <(find . -name "package-info.java" -exec grep -HnE "@(com.yahoo.api.annotations.)?PublicApi.*" {} \; | awk -F/ '{print $2}' | sort -u)); do
```

Comment:
This is not very obscure, but I 100% grant that your implementation is more readable. 👍🏼 

😺